go/libraries/doltcore/merge/merge_prolly_rows.go

```diff
@@ -1148,8 +1150,12 @@ func (m *secondaryMerger) merge(ctx context.Context, diff tree.ThreeWayDiff, sou
 					return fmt.Errorf("cannot merge keyless tables with reordered columns")
 				}
 			} else {
-				valueMappedToMergeSchema := remapTuple(diff.Right, sourceSch.GetValueDescriptor(), m.valueMerger.rightMapping)
-				newTupleValue = val.NewTuple(m.valueMerger.syncPool, valueMappedToMergeSchema...)
+				tempTupleValue, err := remapTupleWithColumnDefaults(ctx, diff.Key, diff.Right, sourceSch.GetValueDescriptor(),
```

Comment:
do we have a test for this one?

Reply:
yup – after I made the change to stop rebuilding secondary indexes as aggressively after a merge, two of our schema merge tests started failing and showed me we had this bug here. Specifically... because we run the schema merge tests left-to-right, and then flip them to run them right-to-left, and because we have the secondary index validation check at the end of our tests, that's what was able to catch this.